File: src/subjects/Subjects.Core/Medication.cs
using System;
using MongoDB.Bson.Serialization.Attributes;
using SharedKernel;

namespace Subjects.Core
{
    public class Medication:Entity
    {
        public string Name { get; set; }
        public string Dosage { get; set; }
        [BsonIgnoreIfNull]
        public DateTime From { get; set; }
        [BsonIgnoreIfNull]
        public DateTime To { get; set; }
        public string Comments { get; set; }
        public Guid Subject { get; set; }
    }
}

File: src/shell/assets/js/subjectApp/models/medication.js
SubjectApp.Medication = DS.Model.extend({
  name: DS.attr('string'),
  dosage: DS.attr('string'),
  from: DS.attr('date', {defaultValue: function(){return new Date();} }),
  to: DS.attr('date', {defaultValue: function(){return new Date();} }),
  comments: DS.attr('string'),
  subject: DS.belongsTo('subject')
});